pvlib/financial.py

```diff
@@ -0,0 +1,146 @@
+import numpy as np
+
+
+def lcoe(production=None, cap_cost=None, fixed_om=None):
+
+    """
+    Real levelized cost of electricity as described on pp.43 and 47-48 by
+    [1](https://www.nrel.gov/docs/legosti/old/5173.pdf)
+    Parameters
+    ----------
+    production : np.array, or pd.Series, default None
+        Annual production [kWh/kW installed]
+    cap_cost : np.array, or pd.Series, default None
+        Initial and annual payments on capital costs using real discount rates
+        [$/kW installed]
+    fixed_om : np.array, or pd.Series, default None
+        Annual payments on operations and maintenance costs [$/kW installed]
+
+    Returns
+    ----------
+    LCOE [cents/kWh]
+
+    References
+    ----------
+    .. [1] W. Short, D. J. Packey, and T. Holt, "A Manual for the Economic
+    Evaluation of Energy Efficiency and Renewable Energy Technologies",
+    NREL/TP-462-5173, 1995.
+    """
+
+    cost = cap_cost + fixed_om
+    return np.nansum(cost)*100/np.nansum(production)
+
+
+def crf(rate, n_years):
+
+    """
+    Capital recovery factor as described on pp. 23 by
+    [1](https://www.nrel.gov/docs/legosti/old/5173.pdf) using real rates
+    Parameters
+    ----------
+    rate : float
+        Real (as opposed to nominal) rate at which CRF is calculated
+    n_years: int
+        Number of years over which CRF is calculated
+
+    Returns
+    ----------
+    crf : float
+        Real capital recovery factor
+    References
+    ----------
+    .. [1] W. Short, D. J. Packey, and T. Holt, "A Manual for the Economic
+    Evaluation of Energy Efficiency and Renewable Energy Technologies",
+    NREL/TP-462-5173, 1995.
+    """
+
+    return (rate*(1+rate)**n_years)/((1+rate)**n_years-1)
```

Comment:
This appears to be different from the "Capital Recovery Factor" defined in the ATB

Reply:
They are algebraically equivalent if you use wacc as the rate in my crf function - to get from the symbolic form defined in the ATB to the symbolic form in mine, just multiply the top and bottom both by (1+wacc)^n